### ISSUE TITLE
Target/Metric fetching should be deferred to the target browser dialog

### ISSUE THREAD

**Issue by fdietz** [fdietz/team_dashboard#38]
Currently, when opening the widget edit dialog (or selecting a different source) we fetch all the available metrics and only after they are fetched we show the actual list of targets.

Instead we want to render the target list right away. Only when opening the target browser dialog we should fetch the list of available metrics. A loading indicator should show the progress.
